actioncable/lib/action_cable/engine.rb

```diff
@@ -22,7 +22,12 @@ class Engine < Rails::Engine # :nodoc:
 
     initializer "action_cable.set_configs" do |app|
       options = app.config.action_cable
-      options.allowed_request_origins ||= /https?:\/\/localhost:\d+/ if ::Rails.env.development?
+      if ::Rails.env.development? && options.allowed_request_origins.nil?
+        local_ips = ["localhost"] + Socket.ip_address_list.select{|ip| ip.ipv4?}.map do |ip|
+          ip.ip_address
+        end
+        options.allowed_request_origins = local_ips.map {|ip| Regexp.new("https?:\/\/#{ip}:\\d+")}
+      end
```

Comment:
Instead of this, what do you think about allowing `config.action_cable.allowed_request_origins` to accept a proc which will evaluate to the expected values as per users need.